gio/src/list_model.rs

```diff
@@ -0,0 +1,31 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::prelude::*;
+use crate::ListModel;
+
+pub trait ListModelExtManual: Sized {
+    // rustdoc-stripper-ignore-next
+    /// Get an immutable snapshot of the container inside the `ListModel`.
+    /// Any modification done to the returned container `Vec` will not be
+    /// reflected on the `ListModel`.
+    fn snapshot(&self) -> Vec<glib::Object>;
+}
+
+impl<T: IsA<ListModel>> ListModelExtManual for T {
+    fn snapshot(&self) -> Vec<glib::Object> {
+        let mut res = Vec::with_capacity(self.n_items() as usize);
+        for i in 0..self.n_items() {
+            res.push(self.item(i).unwrap())
+        }
+        res
+    }
+}
+
+impl std::iter::IntoIterator for ListModel {
+    type Item = glib::Object;
+    type IntoIter = std::vec::IntoIter<glib::Object>;
+
+    fn into_iter(self) -> Self::IntoIter {
```

Comment:
Can't we document the into_iter function here saying it makes use of snapshot function? other than that lgtm

Reply:
I think you can document the `impl` block but not the methods. That documentation is quite invisible in rustdoc though :)

Reply:
> you can document the `impl` block but not the methods

You can also document each method. It'll override the general documentation of the trait.

> That documentation is quite invisible
Indeed, it's not that visible.

In any case, I've added the documentation